### PR TITLE
Learning Twitter’s Bootstrap framework by building a site that actually uses it – Newsweek.

### PR DESCRIPTION
**Pull request by sumon0002001**
*Learning Twitter’s Bootstrap framework by building a site that actually uses it – Newsweek.*

-Used HTML
-CSS is used
-we mainly use the bootstrap framework for creating the layout and styling the page. For example, we used
-Bootstrap grid
-Bootstrap media query
-Bootstrap card
-Bootstrap form for emails.
-Bootstrap buttons.
-Bootstrap content for editing text and other content.

### CODE REVIEW

**Comment by sumon0002001** [comment]
**ADDED HTML, CSS AND BOOTSTRAP**
-HTML
-CSS
-Bootstrap4
This Website is about making a clone page of Newsweek.

**Review by jubaan — CHANGES_REQUESTED** [review]
# Status: Changes Required :recycle: 

## :wave: :wave: Hello @sumon0002001 and @Takhmina175! Thanks for submitting your code!

:100: the website follow the structure of the original
:100: The website has all the elements as the original
:100: The website is responsive

There are some issues that need to be addressed before we can move forward. Please review the list below.

## Required changes :wrench: 
### [Mandatory] :red_circle: 

- [x] Edit the PR message so it contains a descriptive PR.
image

- [x] The PR should be more descriptive than the one you have in a comment below.
image

- [x] Ensure that all linter checks pass before submitting for a re-review. Right now it doesn't pass.

- [x] Improve the commit messages and commit ofter.

- [x] Give one new line space between rules in the CSS file and maintain the consistency through the file

- [x] Use semantic HTML for the main section ```<main>``` as the project uses in the nav and footer.

- [ ] the original doesn't take 100% width in screens larger than 1312px
image

- [ ] Nav doesn't look as original in desktop view
https://microverse.zoom.us/j/5156882051?pwd=UE92N3prSkQxVXBwR0pjeHR5SzFFUT09

- [ ] The debate section doesn't look as expected in desktop view
image

- [ ] Elements doesn't look as expected
image

- [ ] image is distorted.
image

- [ ] Newsletter section doesn't look as expected
image

- [ ] Section doesn't look as expected in tablet view
image

- [ ] Element doesn't look as expected in tablet view
image

- [ ] Element doesn't look as expected
image

- [ ] Element doesn't look as expected
image

- [ ] Navbar doesn't look as original in mobile version
image

- [ ] Navbar overlaps title for Top stories
image

- [ ] Title is not aligned
image

- [ ] Element doesn't looks as original
image

- [ ] Sections missing: Us, World, Business... etc
image

- [ ] email element look distorted.
image


### [OPTIONAL]  :yellow_circle:  


### [Comments] :green_circle: 

- A great tool to test in the website looks as intended in different screen sizes is to use the browsers media query inspector. Please use it to inspect the site and compare it against the original.

## Keep up the great work :muscle: and Happy Coding :smiley: :computer:!!

**Review by jubaan — CHANGES_REQUESTED** [review]
# Status: Changes Required :recycle: 

There are still some issues that need to be addressed before we can move forward. Please review the list below. **Do not submit for a re-review until the changes are made**.

## Required changes :wrench: 
### [Mandatory] :red_circle: 

- [x] Improve the preview screenshot in the readme
image

- [x] Update the title of the PR for something more meaningful and descriptive

- [x] Navbar is missing date in desktop view
image

- [x] Link to live preview is not updated in README

> ## Required changes 
> ### [Mandatory] 
> * [x]  Edit the PR message so it contains a descriptive PR.
>   image
> * [x]  The PR should be more descriptive than the one you have in a comment below.
>   image
> * [x]  Ensure that all linter checks pass before submitting for a re-review. Right now it doesn't pass.
> * [x]  Improve the commit messages and commit ofter.
> * [x]  Give one new line space between rules in the CSS file and maintain the consistency through the file
- [x]  **Use semantic HTML for the main section `<main>` as the project uses in the nav and footer.**
- [x]  **the original doesn't take 100% width in screens larger than 1312px**
>   image
- [x]  **Nav doesn't look as original in desktop view**

- [x]  **The debate section doesn't look as expected in desktop view**
>   image
- [x]  **Elements doesn't look as expected. NOW the element is missing**
>   image
- * [x]  **image is distorted.**
>   image
- [x]  N**ewsletter section doesn't look as expected**
>   image
- [x]  Section doesn't look as expected in tablet view
>   image
- [x]  **Element doesn't look as expected in tablet view**
>   image
- [x]  **Element doesn't look as expected**
>   image
- [x]  **Element doesn't look as expected**
>   image
- [x]  **Navbar doesn't look as original in mobile version**
>   image
- [x]  **Navbar overlaps title for Top stories**
>   image
- [x]  **Title is not aligned**
>   image
- [x]  **Element doesn't looks as original**
>   image
- [x]  **Sections missing: Us, World, Business... etc**
>   image
- [x]  **email element look distorted.**
>   image
> 
> ### [OPTIONAL] 
> ### [Comments] 
> * A great tool to test in the website looks as intended in different screen sizes is to use the browsers media query inspector. Please use it to inspect the site and compare it against the original.
> 
> ## Keep up the great work  and Happy Coding  !!

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

**I must say, you've done an amazing job so far. However, there are some required changes to be fixed before this project can be approved. Please check my points below and fix them as appropriate.**

**[ REQUIRED CHANGES ]**

_Please note that apart from the main point of this project (using bootstrap), this review will be based on the fact that you are asked to be more precise and to make sure that your page looks exactly the same as the source page._

- [x] There is a very glaring difference between the color of your Newsweek header and the original color, as seen in the image below. Please use a similar color.

Original Page:


Your page:


- [x] You have still not updated the title of your Pull Request. This is very important. Check the image below on how to update the title of your PR.

pr

**On medium screen sizes**

- [x] There are some articles after the `Opinion` section, before the `In the magazine` section. These articles should not be here, based on the view from the original page.
- What I'm saying is, for this screen size, the `Opinion` section should be immediately followed by the `In the magazine` section.
- This is also happening at small screen sizes.

Original page:


Your page:


- [x] Since the `Sign up` section highlighted in the image below is present in your project, set it to be in the center.

Original page:


Your page:


- [x] Adjust the height of the newsletter image on the `Newsletter` section of your page.

| Original | Clone (your page) |
| -------------- | --------------- |
|  |  |

**Comments**
- In your README.md file, I can see that the address bar and the taskbar of your screen captured with the screenshot of the project. Just like the last reviewer mentioned, personally, I also think you should crop away your taskbar and the browser address bar from the screenshot.
- We're almost there, do not give up on this project. It's a large project and most of these changes cannot be figured out once because most times, one change leads to another.

**Please make the required changes and submit another review request when you are done.**

* * *

**_You can reach out to me on twitter (@ola_lere), Github(@teekaytech) or slack (@taofeekolalere )_**
__Happy Coding :smiley:__

**Review by ghost — CHANGES_REQUESTED** [review]
# Status: Changes Required

### Hello team,

**Great job implementing the changes. However, there are still a few fixes needed.**

1). **The navigation bar for the desktop screen size**

original | clone
---------| -------
nav_original | nav_clone

- **Please fix your navbar to look like the original.**

- **The navbar background color and Newsweek logo background color should match like in the original.**

- **The two bottom navbars are missing on your page.**

2). **The signup form is no longer showing on your page for desktop screen size(>992px). Please add it. Also, change the design in the medium and small screen sizes so it looks like the original.**

signup_original

3). **A few changes requested by the previous tse have not been implemented yet. Check the screenshot below.**

prev_review


4). **The sections,(u.s, world, business, tech&science, culture and sports), between the editor's pick and subscribe block are missing on your page on small screen sizes.**

5). **Crop out the taskbar and the browser address bar from the readme file screenshot.**

6). **Add the .DS-store in the .gitignore file since it is not related to the project.**

### Happy coding 😃.

**PS: You can reach me on slack (Rose Wanjohi).**

**Review by ghost — CHANGES_REQUESTED** [review]
# Status: Changes Required

## Hello team, 

**Please make the changes requested above. In case of anything, you can reach me on slack (Rose Wanjohi).**

### Happy coding 😃.

**Review by ghost — CHANGES_REQUESTED** [review]
# Status: Changes Required

## Hello team,

**I apologize for my previous review. It was based on the Newsweek branch instead of the feature branch.**

**You have made all changes but there are a few still needed before the project is approved.**

1). **On 992px width, some content in the mid navbar section is displaying on two lines instead of one line like in the original. Example tech & science.**

midnav_clone

2). **The three bottom images in the u.s, world, business, tech&science, culture and sports sections, are not well aligned on small screen sizes. Here is an example.**

us_clone

3). **Implement the following change requested by the previous tse.**

newsletter_clone

4).  **Add the .DS-store file and node_modules folder in the .gitignore file. They are not supposed to be committed as a requirement by Microverse.**

**Review by Rhelli — CHANGES_REQUESTED** [review]
# Changes Required 🛑♻️

Hi guys, excellent job on the project so far. It's fully responsive and stays true to Newsweeks original design. There are however a few outstanding changes that are required in order to approve this project. Let's get them sorted so you can complete this project and move onto the next one!

### Mandatory Changes
 1. **[Mandatory]** - At present, your opinions section appears to flow straight into a duplicated section instead of 'More Stories'.

**Newsweeks website** 👇🏽

image

**Your version of Newsweek** 👇🏽

image

**Fix:** Your version of Newsweek _does_ have the 'More Stories' section, all you need to do is get rid of this duplicated section. This is appearing on small screen sizes.

******
 2. **[Mandatory]** - The image in the following 'Sign Up' area is broken and the `alt` text is displaying where it should not.

**Here is your version** 👇🏽
image

**Here is Newsweeks version** 👇🏽
image

**Fix:** In your HTML (line 321) you have wrapped the `img` tag inside of a `label` tag. I suspect if you separate them, it will solve this (if it is not an error to do with the path to the image or an error with the image itself)

******

3. **[Mandatory]** - Your subscribe section images and texts are overlapping each other making the text unreadable. This occurs on both small and medium-size screens

**Small Screen Size** 👇🏽
image

**Medium Screen Size** 👇🏽
image

**Fix:** Place the text in its own container and place it beneath the magazine images. If it is still overlapping, check the entire subscribe container height to ensure it is not squeezing all its contents into too small a space (causing overlapping).

*****

4. **[Mandatory]** - Please check the text alignment with the images below

**Your version of Newsweek** 👇🏽
image


**Newsweeks version** 👇🏽
image


**Fix:** Convert (if not already) each container holding the image and the headline to flex. Then give that container the property of `flex-direction: row`. That should ensure the text aligns to the right of the image, not beneath it.

****** 

As I said, an excellent job so far on the project, so don't be disheartened by the requested changes. Get them fixed and you'll be on to the next project in no time!

If you have any extra questions or need any help, @ me here on Github or get in contact with me via Slack and I'll do what I can to help.

All the best,

Rory

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
### Request changes 😳
I am very happy to see this project and I can see you worked hard on this project but I am sorry to inform you that I cannot approve this project because it is not completed yet. You need to work more on it. 
_It's okay. We are here to learn. Don't get discouraged. Work even harder. You are almost there.😄_
**Please have a look at the suggestions to get more understanding.**

- [ ] Please update the title of PR. 
95893380-ca3dfd80-0d7f-11eb-922b-50e6d0a75dc3

- [ ] Fix: Convert (if not already) each container holding the image and the headline to flex. Then give that container the property of flex-direction: row. That should ensure the text aligns to the right of the image, not beneath it.


- [ ] top navbar should be sticky and decrease its width on scrolling. 

**Clone**

chrome-capture (31)

**Original**

chrome-capture (32)

- [ ] In editor's pick every item is aligned in a row but on your clone,

**Clone**

chrome-capture (32)
 there is an alignment error. Please have a look below. 

**Original**

chrome-capture (33)

- [ ] Width of left side items should be equal to right side items. Please have a look below. 

**Clone**

chrome-capture (34)

**Original**

chrome-capture (35)

- [ ] Newsletter and subscribe section should have equal width. Please have a look below. 

**Clone**

chrome-capture (36)


**Please make the changes and request re-review for the given project**
Stay safe and give me a smile 😇

### Happy Coding
---------------------------------------------------------------------------------------------------------------------------------
**Regards,
Arslan Bisharat
slack: arslanbisharat**

File: index.html
Comment (arslanbisharat) [inline index.html:1020]: logo Image is not working

File: index.html
Comment (arslanbisharat) [inline index.html]: Please add padding around these and align them correctly as present on the original page.

File: index.html
Comment (arslanbisharat) [inline index.html]: Please add padding around these and align them correctly as present on the original page.

Comment (arslanbisharat) [inline index.html]: Image is not working

File: index.html
Comment (arslanbisharat) [inline index.html]: Please add padding around these and align them correctly as present on the original page.

File: index.html
Comment (arslanbisharat) [inline index.html:979]: Please add padding around these and align them correctly as present on the original page.

File: index.html
Comment (arslanbisharat) [inline index.html:973]: Please add padding around these and align them correctly as present on the original page.

File: index.html
Comment (arslanbisharat) [inline index.html:955]: Image not working

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
### Request changes 😳
I am very happy to see this project and I can see you worked hard on this project but I am sorry to inform you that I cannot approve this project because it is not completed yet. You need to work more on it. 
_It's okay. We are here to learn. Don't get discouraged. Work even harder. You are almost there.😄_
**Please have a look at the suggestions to get more understanding.**

- [ ] Please change **Learning Twitter’s Bootstrap framework by building a site that actually uses it – Newsweek. #1** to  **Building with Bootstrap** in your PR Title 

- [ ] As described earlier, please make the navbar sticky. Please scroll the [original page](https://www.newsweek.com/) to get more understanding.  Here is [resource 1](https://www.w3schools.com/howto/howto_js_navbar_sticky.asp) and [resource 2](https://bootsnipp.com/snippets/E10d8)

- [ ]  The images in the webpage are not rendering.
 
chrome-capture (40)

chrome-capture (38)
chrome-capture (39)

- [ ] As I cannot see the images, I cannot review your project completely. Please resolve that issue. And most importantly work on all the changes requested in previous reviews. 

**Please make the changes and request re-review for the given project**
Stay safe and give me a smile 😇

### Happy Coding
---------------------------------------------------------------------------------------------------------------------------------
**Regards,
Arslan Bisharat
slack: arslanbisharat**

**Review by elmejdki — APPROVED** [review]
# status: approved :heavy_check_mark: 
## Well done @team :clap: :clap: :clap:, thanks for making the required changes :toolbox: by the previous TSE's.

now you can merge, your project is approved.

## Happy Coding :tophat:

**Pull request merged**
